### ISSUE TITLE
MySQL doesn't support `default` on `json` columns.

### ISSUE THREAD

**Issue by andrewculver** [bullet-train-co/bullet_train-roles#4]
Instead, I think we'll need to remove the attempt to set a `default` on the column when we generated the migration, and instead do this in the model:

```
after_initialize :set_default_role_ids

def set_default_role_ids
  self.role_ids ||= []
end
```

**Comment by coolprobn** [comment]
@andrewculver I think we can add default value for postgresql databases while we add the solution you have suggested for mysql databases? 

We are already checking database type so just skipping for mysql won't be much of a work. I also think that if we can take advantage of database functions like default values then Rails will have one less thing to do in app with postgresql databases which is a good thing.

**Comment by adampal** [comment]
@andrewculver @coolprobn I had to address this issue today because I needed to support a mysql database.
I've updated the gem to handle nil values in the role_ids column and removed the default value so we're handling all databases in the same way.
Technically, I think this issue is now closed.  However it doesn't address @coolprobn's concern about making Rails do extra work that is unnecessary when a default value is available.  
My feeling is that it is probably immeasurably small the performance difference when doing `nil || []` in a couple of places and we get to reduce complexity in the generator and handle all databases in the same manner, I thought it was an appropriate solution.  Happy to discuss if either of you feel differently though.
I'll close this one for now but we can re-open if you think it's worth continuing.
